### PR TITLE
Run ConnectS2I with tini

### PR DESCRIPTION
**Pull request by stanlyDoge**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Bugfix

### Description
The ConnectS2I deployment is not run with `tini` as PID 1. It is caused by the composition of scripts. Connect S2I is run by `/docker-images/kafka/s2i-scripts/run` which runs `/opt/kafka/kafka_connect_run.sh` which runs `tini`. So first process in Connect is `tini`, but in ConnectS2I it is `bash`.

After the change:
```
[sknot@localhost productizedf8]$ oc exec -ti my-s2i-connect-cluster-connect-1-7wllq -- ps aux
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
1000150+     1  0.0  0.0   4360   360 ?        Ss   10:00   0:00 /usr/bin/tini -w -e 
1000150+    31 47.0  3.5 8166700 578940 ?      Sl   10:00   0:34 java -Xms128M -serve
1000150+   511  0.0  0.0  51756  1656 ?        Rs+  10:01   0:00 ps aux

```

One question for @im-konge : Why is test `testCheckContainersHaveProcessOneAsTini` in LogSettingST class? It does not test any logging or?

Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/3282
### Checklist

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by im-konge** [comment]
@stanlyDoge we were finding best place where to put it and where all the resources are deployed, so I decided to choose the `LogSettingST`, but I'll do some changes and it will be in another suite ... let's keep it there before the "small" refactor.

**Review by im-konge — APPROVED** [review]

File: docker-images/kafka/s2i-scripts/run [hunk 1]
```diff
@@ -11,5 +11,6 @@ then
 fi
 
 export KAFKA_CONNECT_PLUGIN_PATH=/tmp/kafka-plugins
+export KAFKA_CONNECT_S2I=true
 echo "Starting Kafka Connect with custom plugin directory $KAFKA_CONNECT_PLUGIN_PATH"
-/opt/kafka/kafka_connect_run.sh
\ No newline at end of file
+exec /usr/bin/tini -w -e 143 -- /opt/kafka/kafka_connect_run.sh
```
Comment (scholzj) [inline docker-images/kafka/s2i-scripts/run:16]: Can't you source the script instead?

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
The code looks good so I approved it. But you will need to fix the shellcheck issues.

**Review by ppatierno — APPROVED** [review]

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=ConnectS2IST profile=regression

**Pull request merged**